src/utils/function.js

```diff
@@ -44,11 +44,50 @@ export function rateLimit(win, callback, minInterval) {
     }
   }
 
-  return function() {
+  return function(...args) {
     if (locker) {
-      nextCallArgs = arguments;
+      nextCallArgs = args;
     } else {
-      fire(arguments);
+      fire(args);
+    }
+  };
+}
+
+/**
+ * Wraps a given callback and applies a wait timer, so that minInterval
+ * milliseconds must pass since the last call before the callback is actually
+ * invoked.
+ *
+ * @param {!Window} win
+ * @param {function()} callback
+ * @param {number} minInterval the minimum time interval in millisecond
+ * @returns {function()}
+ */
+export function debounce(win, callback, minInterval) {
+  let locker = 0;
```

Comment:
`setTimeout` returns a positive int, so I use 0 to mean no timeout scheduled. Just keeps the type the same in the JS engine.